entity-service-impl/src/test/java/org/hypertrace/entity/query/service/converter/filter/NullFilteringExpressionConverterTest.java

```diff
@@ -0,0 +1,129 @@
+import org.hypertrace.entity.query.service.converter.Converter;
+import org.hypertrace.entity.query.service.converter.identifier.IdentifierConverter;
+import org.hypertrace.entity.query.service.converter.identifier.IdentifierConverterFactory;
+import org.hypertrace.entity.query.service.v1.ColumnIdentifier;
+import org.hypertrace.entity.query.service.v1.LiteralConstant;
+import org.hypertrace.entity.query.service.v1.Operator;
+import org.hypertrace.entity.query.service.v1.Value;
+import org.hypertrace.entity.query.service.v1.ValueType;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+class NullFilteringExpressionConverterTest {
+  private static final RequestContext REQUEST_CONTEXT = RequestContext.forTenantId("tenant1");
+
+  @Mock private EntityAttributeMapping entityAttributeMapping;
+  @Mock private IdentifierConverterFactory identifierConverterFactory;
+  @Mock private Converter<LiteralConstant, ConstantExpression> constantExpressionConverter;
+
+  private NullFilteringExpressionConverter nullFilteringExpressionConverter;
+
+  @BeforeEach
+  void setup() {
+    MockitoAnnotations.initMocks(this);
+
+    nullFilteringExpressionConverter =
+        new NullFilteringExpressionConverter(
+            entityAttributeMapping, identifierConverterFactory, constantExpressionConverter);
+  }
+
+  @Test
+  void testEqNull() throws ConversionException {
+    ColumnIdentifier columnIdentifier =
+        ColumnIdentifier.newBuilder().setColumnName("column1").build();
+    LiteralConstant constant =
+        LiteralConstant.newBuilder().setValue(Value.newBuilder().setString("null").build()).build();
+
+    when(entityAttributeMapping.getDocStorePathByAttributeId(REQUEST_CONTEXT, "column1"))
+        .thenReturn(Optional.of("subDocPath1"));
+
+    IdentifierConverter identifierConverter = mock(IdentifierConverter.class);
+    when(identifierConverter.convert(any(), eq(REQUEST_CONTEXT)))
+        .thenReturn("attributes.subDocPath1");
+    when(identifierConverterFactory.getIdentifierConverter(
+            "column1", "subDocPath1", ValueType.STRING, REQUEST_CONTEXT))
+        .thenReturn(identifierConverter);
+
+    ConstantExpression constantExpression = ConstantExpression.of("null");
+    when(constantExpressionConverter.convert(constant, REQUEST_CONTEXT))
+        .thenReturn(constantExpression);
+
+    FilterTypeExpression filterTypeExpression =
+        nullFilteringExpressionConverter.convert(
+            columnIdentifier, Operator.EQ, constant, REQUEST_CONTEXT);
+    IdentifierExpression identifierExpression = IdentifierExpression.of("attributes.subDocPath1");
+
+    assertEquals(
+        LogicalExpression.builder()
+            .operator(LogicalOperator.OR)
+            .operands(
+                List.of(
+                    RelationalExpression.of(
+                        identifierExpression, RelationalOperator.NOT_EXISTS, constantExpression),
```

Comment:
there shouldn't be a RHS of a not exists expression - is this being used or just a bad API?

Reply:
Just a bad API. It isn't being used on the document store implementation side